contracts/RiskFund/ReserveHelpers.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../ComptrollerInterface.sol";
+
+contract ReserveHelpers {
+    using SafeERC20 for IERC20;
+
+    // Store the previous state for the asset transferred to ProtocolShareReserve combined(for all pools).
+    mapping(address => uint256) internal assetsReserves;
+
+    // Store the asset's reserve per pool in the ProtocolShareReserve.
+    // Comptroller(pool) -> Asset -> amount
+    mapping(address => mapping(address => uint256))
+        internal poolsAssetsReserves;
+
```

Comment:
We should have a gap here, I guess – just in case we need to add another storage var and not ruin the storage layout:

```
uint256[48] __gap;
```
(48 is because the convention is to have 50 slots, and you have 2 slots occupied by mappings)